notifications_utils/field.py

```diff
@@ -63,8 +63,8 @@ class Field:
     """
 
     placeholder_pattern = re.compile(
-        r'\({2}'        # opening ((
-        r'([^()]+)'     # body of placeholder - potentially standard or conditional.
+        r'\({2}(?=[^(])'        # opening ((
+        r'.+?(?=\)\))'     # body of placeholder - potentially standard or conditional.
```

Comment:
Minor: we could use `\){2}` on the end, to match the opening regex.

Reply:
We can simplify this to just `.+?`. It's also worth adding a comment about the purpose of the optional / non-greedy `?` modifier.

Reply:
>  Minor: we could use \){2} on the end, to match the opening regex.

Done now ☺️ 

> We can simplify this to just .+?. It's also worth adding a comment about the purpose of the optional / non-greedy ? modifier.

Sorry, I don't understand what you want me to do here.

Reply:
Haha, neither did I at first 😆. After reading through it again, I'm asking for two things:

- We can change `r'.+?(?=\)\))'` to just `r'.+?'` (all the tests still pass).
- Extend the comment to explain why we need the `?` in `.+?`.

The last point is because I found it hard to understand what the `?` is doing: the tests fail without it, but it wasn't clear why. I then found this in the docs: `Adding ? after the qualifier makes it perform the match in non-greedy or minimal fashion`. So without it we'd match `((something)))` from `text with (((something))) in brackets` instead of the correct `((something))`. It has a similar, but opposite, purpose to your change on line 66.

Does that help?

Reply:
Wow, you are right, the simpler regex works, too, amazing! 💖 

I added an explanation about the question mark in the commit message. I feel it is a bit too lengthy and technical for an inline comment.